src/services/atomic-swap.service.js

```diff
@@ -29,6 +29,17 @@ const {
  */
 const walletListenedProposals = new Map();
 
+/**
+ * Represents errors related to interactions with the Atomic Swap Service
+ */
+class SwapServiceError extends Error {
```

Comment:
I think would be better to have a separate file with the errors (and with a TODO to use the errors from the lib when you add it there)

And why don't you create a simple Error class like this?

```
class SwapServiceError extends Error {}
```

Reply:
This code was successfully rebased from 286